File: csharp/demo/demo01ex-name.cs
/*
 * HELLO WORLD VERSION MULTITHREADING
 * Version extra: Getting thread's name
 */
using System;
using System.Threading;



class Demo01ExtraName : IRunnable
{
    public void run()
    {
        Thread thFoo = new Thread(doTask)
        {
            Name = "foo"
        };

        Thread thBar = new Thread(doTask);
        thBar.Name = "bar";

        thFoo.Start();
        thBar.Start();
    }

    private void doTask()
    {
        Console.WriteLine($"My name is {Thread.CurrentThread.Name}");
    }
}

File: python/demo01ex_name.py
'''
HELLO WORLD VERSION MULTITHREADING
Getting thread's name
'''

import threading



def do_task():
    print(f'My name is {threading.current_thread().name}')



th_foo = threading.Thread(target=do_task, name='foo')
th_bar = threading.Thread(target=do_task, name='bar')
th_foo.start()
th_bar.start()
